Dockerfile

```diff
@@ -1,78 +1,138 @@
-FROM python:3.7.1-stretch
+# Dockerfile contains multiple levels of images--
+# First, a base image containing dependencies shared by all other images:
+#   - apt dependencies
+#   - rust
+#   - ntl
+#   - pbc
+#   - charm
+#   - base pip dependencies (cython & setup.py)
+# 
+# Thereafter, it add ever increasing levels of dependencies-- 
+#   - Test requirements (including doc requirements)
+#   - Dev requirements (including aws)
+#
+# In order to build and push this to dockerhub, run:
+# docker build . --target test-image --tag dsluiuc/honeybadgermpc-docker-base
+# docker push dsluiuc/honeybadgermpc-docker-base:latest
+
+
+FROM python:3.7.3-slim AS base-image
```

Comment:
Neat! What's the size of the new image?

Reply:
It starts really small... then when everything else gets installed to get the tests running, we sit at a bit under 2GiB